### ISSUE TITLE
display manager (PCDM): the keyboard is mostly ineffective until after a press on the Tab key

### ISSUE THREAD

**Issue by grahamperrin** [trueos/trueos-core#266]
This issue became most noticeable after I preferred 
`autoboot_delay="15"` in 
`/boot/loader.conf` and it feels regressive. 

At the FreeBSD loader: around the end of the fifteen second countdown, when it reaches '2', the keyboard is (still) not driven. 

Also after my most recent log out from KDE, there was (if I'm not mistaken) a remarkable delay – at the display manager – before the OS responded properly to the keyboard. 

```
$ uname -v ; uname -n
FreeBSD 12.0-CURRENT #12 5ef9376(drm-next-4.7): Thu Dec 29 16:26:36 UTC 2016     root@gauntlet:/usr/obj/usr/src/sys/GENERIC 
momh167-gjp4-hpelitebook850g2-trueos.university.brighton.ac.uk
$ 
```
I suspect that the problem began after an update from `5ef9376` to `5ef9376` (an update, but no change to the commit reference). 

## Environment

- HP EliteBook 850 G2
- Logitech mouse at the rearmost USB port on the left of the notebook
- Kensington keyboard at the frontmost port on the left. 

```
$ cat /etc/rc.conf
rc_verbose="YES"
rc_interactive="YES"
rc_depend_strict="YES"
rc_parallel="NO"
rc_crashed_start="YES"
rc_logger="YES"
rc_log_path="/var/log/rc.log"
zfs_enable="YES"
kldload_i915kms="i915kms"
kldload_vbox="vboxdrv"
dbus_enable="YES"
# https://github.com/trueos/freebsd/commit/0af2dbd9b11c7ac54baef4b3bcd762f85bd3a658
linux_enable="YES"
wlans_iwm0="wlan0"
# wlans_rum0="wlan0"
ifconfig_wlan0="WPA DHCP"
ifconfig_em0="193.62.168.100 netmask 255.255.255.0"
defaultrouter="193.62.168.1"
hostname="momh167-gjp4-hpelitebook850g2-trueos.university.brighton.ac.uk"
sshd_enable="YES"
vboxnet_enable="YES"
kldload_nvidia="nvidia-modeset nvidia"
ifconfig_em0_ipv6="inet6 accept_rtadv"
$ 
```

```
$ cat /boot/loader.conf
# * IMPORTANT NOTICE *
# If using GRUB, run 'grub-mkconfig -o /boot/grub/grub.cfg' after making
# changes to this file
###############################################################################
crypto_load="YES"
aesni_load="YES"
geom_eli_load="YES"
# Tune ZFS Arc Size - Change to adjust memory used for disk cache
vfs.zfs.arc_max="8192M"
zfs_load="YES"
# boot_mute="NO"
autoboot_delay="15"
$ 
```

```
$ cat /boot/loader.conf.trueos 
# Kernel Options
kern.ipc.shmseg=1024
kern.ipc.shmmni=1024
kern.maxproc=10000

# Accept the wifi firmware license
legal.intel_ipw.license_ack=1
legal.intel_iwi.license_ack=1
legal.realtek.license_ack=1

# Load File-System Support
opensolaris_load="YES"
zfs_load="YES"
tmpfs_load="YES"

# Disable acpi throttle by default, fixes issues with powerd and later AMDs
hint.acpi_throttle.0.disabled="1"

# Show keystokes for passphrases
kern.geom.eli.visible_passphrase=2 

# Speed up boot
kern.cam.scsi_delay="500"

# Skip the often slow memory tests in VMs
hw.memtest.tests=0

# Load geom support
crypto_load="YES"
geom_eli_load="YES"

# Enable hardware crypto
aesni_load="YES"

# Enable NewCons / vt
kern.vty=vt

# Disable some labels
kern.geom.label.gptid.enable="0"
kern.geom.label.disk_ident.enable="0"
kern.geom.label.ufsid.enable="0"

# Don't power devices with no driver
hw.pci.do_power_nodriver=3

# Disable the ipv6 auto_linklocal.
# If you use the lagg interface you may want to set this in /boot/loader.conf
#net.inet6.ip6.auto_linklocal="0"

# Set ipfw to default accept
net.inet.ip.fw.default_to_accept="1"

# Allow more than one pass through firewall (for pipes, etc)
net.inet.ip.fw.one_pass=0

# Increase maxdata for net.graph, fixes issues with scp / ssh wile running VirtualBox
net.graph.maxdata=65536

# Set the brand/loader logo
loader_brand="trueos"
# loader_logo="trueos"
loader_menu_title="catch me while you can"

# Enable Synaptics for touchpads
hw.psm.synaptics_support="1"

# Enable Elantech touchpad support
hw.psm.elantech_support="1"

# Speed up boot time
autoboot_delay="01"

# Enable the TrueOS splash-screen
# boot_mute="YES"
$ 
```

[pciconf -lv.txt](https://github.com/trueos/trueos-core/files/680102/pciconf.-lv.txt)

**Comment by grahamperrin** [comment]
## At the display manager screen, after log out

Not limited to KDE; symptoms here are also reproduced following every log out from Lumina. 
Most remarkable: 

- keying Caps Lock results in toggles of the (LED) indicator within the keyboard, proving that the OS is in some way responsive to the HID
- with a list of three users, keying down has no effect
- with that list of three users, keying Escape has no effect (no prompt to shut down the OS). 

### Workaround

- key Tab.

## At the FreeBSD loader

The tab key is not a workaround. 

No visible response, at the indicator within the keyboard, to keying Caps Lock. It's as if truly, the keyboard is not driven in good time. 

I wonder whether we have two separate issues here.

**Comment by pkgdemon** [comment]
Not released yet.  Not even pushed to UNSTABLE.

**Comment by grahamperrin** [comment]
(Understood, thanks … and more recently reproduced with a different version of `FreeBSD 12.0-CURRENT #12 5ef9376(drm-next-4.7): Thu Dec 29 16:26:36&nbsp;…`)

**Comment by pkgdemon** [comment]
Reopening after push to unstable

**Comment by pkgdemon** [comment]
Just a thought.  We could also consider adding this to the Post Installation Tasks screen, and call that screen something else besides optional services.  Unless it may be more desirable to have right click all the way through the installation as well.

https://www.trueos.org/handbook/postinstall.html#config6

**Comment by pkgdemon** [comment]
Posted to wrong ticket the first time.  Ignore deleted message.

**Comment by grahamperrin** [comment]
> I wonder whether we have two separate issues here.

We probably do. 

----

> ## Keyboard on USB apparently not driven in good time at startup, …

First impressions of trueos-desktop-201701060022 on `FreeBSD 12.0-CURRENT #13 42158f1(drm-next-4.7): Thu Jan  5 22:37:19 UTC 2017     root@gauntlet:/usr/obj/usr/src/sys/GENERIC`: 

- probably fixed

– I'll continue to test. If reproducible, I should probably create a separate issue. 

----

> ## Keyboard on USB … only partially usable when the display manager reappears after log out

… and when the manager first appears. 

Early impressions of trueos-desktop-201701060022 on `FreeBSD 12.0-CURRENT #13 42158f1(drm-next-4.7): Thu Jan  5 22:37:19 UTC 2017     root@gauntlet:/usr/obj/usr/src/sys/GENERIC`: 

- the issue is reproducible at four of four notebooks. 

Most easily reproducible wherever the manager lists **two or more users**: 

- keying Return or Enter fails to present the authentication dialogue for the selected user.

**Comment by grahamperrin** [comment]
With the change of title of this issue (reduction from two issues, one of which was transient, to one): 

- is the issue reproducible at iXsystems? 

## display manager (PCDM): the keyboard is mostly ineffective until after a press on the Tab key

The workaround (Tab) is trivially easy when known, but not intuitive when unknown. 

If this issue creeps into STABLE then some customers may be frustrated by the sense of falling at the first hurdle. No response to Return or Enter, when a user's name is selected, feels like an obstacle.

**Comment by grahamperrin** [comment]
This seems to be fixed in _yesterday's_ `FreeBSD 12.0-CURRENT #20 3dbce2a(drm-next): Sat Jan 21 13:48:35 UTC 2017     root@gauntlet:/usr/obj/usr/src/sys/GENERIC` (sorry, I forgot to run `about` to better identify the installation). 

In which repo might I find the commit?

**Comment by beanpole135** [comment]
https://github.com/trueos/pcdm/commit/c66a6648d1a913059d716fdc03d6fc4ebc30f369
